packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+
+const DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY = 'DISMISSED_PROMO_BANNERS';
+@Component({
+  tag: 'va-promo-banner',
+  styleUrl: 'va-promo-banner.css',
+  shadow: true,
+})
+export class VaPromoBanner {
+  @Element() el: HTMLElement;
+
+  /**
+   * The rendering custom markup via the slot instead of the `<a>` with `text` in it
+   * */
+  @Prop() renderCustom: boolean = false;
+
+  /**
+   * `href` attribute for the `<a>` tag. Only gets used if `render` is _not_ used
+   */
+  @Prop() href: string;
+
+  /**
+   * `target` attribute for the `<a>` tag. Only gets used if `render` is _not_ used
+   */
+  @Prop() target: string;
+
+  /**
+   * Controls which icon gets used. Options are 'announcement', 'news', or 'email-signup'.
+   * */
+  @Prop() type: 'announcement' | 'news' | 'email-signup';
+
+  /**
+   * Analytics tracking function(s) will not be called
+   */
+  @Prop() disableAnalytics: boolean;
+
+  /**
+   * Id of Promo Banner Instance for tracking in Storage
+   */
+  @Prop() bannerId: string;
```

Comment:
What might this be, and do we need a separate prop for it or could we just use the element's `id`, or some combination [like we're doing in `<va-banner>`](https://github.com/department-of-veterans-affairs/component-library/blob/3775aeb5fb3ea3f8d3ea26dd57afc5fb268d096b/packages/web-components/src/components/va-banner/va-banner.tsx#L71-L72)?

Reply:
So its coming from here: https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/components/banners.drupal.liquid#L33
So we are actually getting an ID from Drupal in this case where as in the Banner we are making it up on the fly in the component.
For that I think we would need to keep it because we would need to map to it but correct my if you have other thoughts happy to update

Reply:
update to this.el.id to do the matching on the banner for storage

Reply:
@bkjohnson I updated it to use the native `id` attribute and remove `render-custom`